### PR TITLE
filter CC appts

### PR DESCRIPTION
**Pull request by jenniemc**
## Description
Vista scheduled CC appts should display it as Community Care appointment and exclude all VAMC related information

## Testing done
local and unit test

## Screenshots
image

image

## Acceptance criteria
- [ ] VistA CC appointments displays Community Care in the header
- [ ] VistA CC appointments should not display any clinic, facility, or provider information
- [ ] VistA CC appointments should have a Cancel appointment link that looks like other CC appointments, but without provider information

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by jenniemc** [comment]
@lenaecb @peggygannon @jbalboni 
Results of removing the facility information, provider information and directional link from Vista generated CC appt are captured in the screenshots in the PR. Not surprisingly, it looks rather "bare"

**Comment by lenaecb** [comment]
@jenniemc - Looks like I didn't get this into the AC but should have.

For these appts, we need to provide the following information...

"This appointment is scheduled with a community care provider. Please do not report to your local VA facility. If you have questions, please contact [your facility](https://www.va.gov/find-locations/) community care staff at your local VA. " -- @peggygannon - any slight iteration to this? This was previously 'approved' language by OVAC for these types of appts.

File: src/applications/vaos/appointment-list/components/cards/confirmed/ConfirmedAppointmentListItem.jsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ export default function ConfirmedAppointmentListItem({
   const isInPersonVAAppointment = !isVideo && !isCommunityCare;
 
   const showInstructions =
-    isCommunityCare ||
+    (isCommunityCare && appointment.contained[0].address) ||
```
Comment (jbalboni) [inline src/applications/vaos/appointment-list/components/cards/confirmed/ConfirmedAppointmentListItem.jsx:54]: This doesn't get the correct result on the page, but it feels like a rule that will be confusing later, since it reads like "we don't show instructions for community care appointments that don't have an address." Which is incidentally true, but not really the root of what we're trying to do.

I wonder if it would be better to exclude these comments in the transformer? We could have some logic in here to only use `instructionsToVeteran` for `comment` for CC appts.

https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/vaos/services/appointment/transformers.js#L525

File: src/applications/vaos/services/appointment/transformers.js [hunk 2]
```diff
@@ -25,10 +25,13 @@ function getAppointmentType(appt) {
     return APPOINTMENT_TYPES.ccRequest;
   } else if (appt.typeOfCareId) {
     return APPOINTMENT_TYPES.request;
-  } else if (appt.clinicId || appt.vvsAppointments) {
-    return APPOINTMENT_TYPES.vaAppointment;
-  } else if (appt.appointmentTime) {
+  } else if (appt.distanceEligibleConfirmed || appt.communityCare === true) {
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/transformers.js:28]: Do we need to check `distanceEligibleConfirmed` as well as the `communityCare` property?

File: src/applications/vaos/services/appointment/transformers.js [hunk 3]
```diff
@@ -164,7 +167,10 @@ function getConfirmedStatus(appointment, isPast) {
  * @returns {Object} Returns appointment datetime as moment object
  */
 function getMomentConfirmedDate(appt) {
-  if (isCommunityCare(appt)) {
+  if (
+    isCommunityCare(appt) &&
+    (appt.communityCare === undefined || appt.communityCare === null)
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/transformers.js:172]: Maybe we should turn this whole check into `if (appt.appointmentTime) {`? While this check is correct, I think it's somewhat confusing ("if this appt is community care but also not community care").

File: src/applications/vaos/components/FacilityAddress.jsx [hunk 4]
```diff
@@ -17,14 +17,31 @@ export default function FacilityAddress({
...
+              <br />
+            </React.Fragment>
+          ))}
+          {address.city}, {address.state} {address.postalCode}
+          <br />
+        </>
+      )}
+      {!address && (
+        <>
+          This appointment is scheduled with a community care provider. Please
```
Comment (jbalboni) [inline src/applications/vaos/components/FacilityAddress.jsx:34]: This `FacilityAddress` component is used in a few different places, not always just on the confirmed appointments card. So I'm not sure we should put this language in here. I think it might make more sense in `ConfirmedCommunityCareLocation`, where we could put some logic that renders `FacilityAddress` if we have an address, and this text if we do not.

**Review by jbalboni — COMMENTED** [review]
Looks good so far. I made some comments about moving around some of the logic, but it looks like all the output is correct.

**Comment by lenaecb** [comment]
@jenniemc - I just realized my mistake. I think we need to move the link to the facility locator to the end of the sentence. 

So - "This appointment is scheduled with a community care provider. Please do not report to your local VA facility. If you have questions, please contact your facility community care staff at [your local VA](https://www.va.gov/find-locations/). "

Gosh, so sorry about that.

**Comment by jenniemc** [comment]
Hi @lenaecb,
Got a question. The alert titles for CC appt cancellation is "You need to call your community care **provider** to cancel this appointment"  but for Vista CC appt cancellation, you want to use the title "You need to call your community care **staff at your local VA facility** to cancel this appointment". Is that correct?

As for the other the changes, not a problem. Thanks for supplying the new text so the hyperlink is at the end of the sentence.

**Comment by lenaecb** [comment]
@jenniemc - yes, I want to try and keep the language consistent where we can. But as you point out, to do that for VistA CC appts we have to deviate just a little bit to support them calling the cc staff at a VA.

File: src/applications/vaos/appointment-list/components/cards/confirmed/ConfirmedAppointmentListItem.jsx (hunk 1 above)
Comment (jenniemc) [reply src/applications/vaos/appointment-list/components/cards/confirmed/ConfirmedAppointmentListItem.jsx]: Good idea about using the comment field in the transformer. I placed a ternary statement to check for `communityCare` so the alert text is passed to the `communityCareInstructions` component

File: src/applications/vaos/services/appointment/transformers.js (hunk 2 above)
Comment (jenniemc) [reply src/applications/vaos/services/appointment/transformers.js]: I switched back to `appointmentTime` instead of `distanceEligibleConfirmed` for the conditional check

File: src/applications/vaos/services/appointment/transformers.js (hunk 3 above)
Comment (jenniemc) [reply src/applications/vaos/services/appointment/transformers.js]: Haha, It does sound confusing when you lay it out in plain English. I'm now checking against `app.timeZone` instead

File: src/applications/vaos/components/FacilityAddress.jsx (hunk 4 above)
Comment (jenniemc) [reply src/applications/vaos/components/FacilityAddress.jsx]: That makes perfect sense. The alert text is now placed in the `communityCareInstructions` component

File: src/applications/vaos/services/appointment/transformers.js [hunk 5]
```diff
@@ -538,7 +540,9 @@ export function transformConfirmedAppointments(appointments) {
       minutesDuration,
       comment:
         appt.instructionsToVeteran ||
-        appt.vdsAppointments?.[0]?.bookingNote ||
+        (appt.communityCare
+          ? 'vistaCCAppt'
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/transformers.js:544]: I think this will put `vistaCCAppt` in the instructions field for regular CC appointments that don't have instructions, which I don't think is what we want. We might want to break this into a separate function to be able to write the logic more clearly.

That said, I'm not positive we want to put a special value in the instructions here, since I don't think the design calls for putting the text in the right hand column like other instructions. I would check with Lenae on that.

**Comment by jenniemc** [comment]
Hi @jbalboni, all the tests passed on the local build but had failed in Jenkins. I learn from this morning not to get latest from master into my working branch. In the meantime, would you review the PR?

**Review by jbalboni — APPROVED** [review]
Looks good to me!

**Comment by jenniemc** [comment]
@jbalboni, I see there are 2 files that have merge conflicts but won't let me resolved using the web editor. It does show steps using the command line. Would you recommend taking those command line steps in terminal?

File: src/applications/vaos/services/appointment/transformers.js [hunk 6]
```diff
@@ -25,9 +25,12 @@ function getAppointmentType(appt) {
     return APPOINTMENT_TYPES.ccRequest;
   } else if (appt.typeOfCareId) {
     return APPOINTMENT_TYPES.request;
-  } else if (appt.clinicId || appt.vvsAppointments) {
+  } else if (
+    appt.vvsAppointments?.length ||
+    (appt.clinicId && appt.communityCare === false)
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/transformers.js:30]: We need to change `appt.communityCare === false` to `!appt.communityCare`. Sometimes this value can come back null.

**Review by jbalboni — CHANGES_REQUESTED** [review]
Found a bug we need to fix

File: src/applications/vaos/services/appointment/transformers.js (hunk 6 above)
Comment (jenniemc) [reply src/applications/vaos/services/appointment/transformers.js:30]: will do

**Comment by jbalboni** [comment]
@jenniemc This is what I normally do:

```
git checkout master
git pull
git checkout feature/12735-Fix-display-of-VistA-Community-Care-appoinments 
git merge master
```

I'm not sure what merge tool that will open, though. You should be able to do this through sourcetree, too.

**Review by jbalboni — APPROVED** [review]
